script.js

```diff
@@ -1,4 +1,134 @@
+//define vars
+var indivisualDiceRoll;
+var mode = `instructions`;
+var playerNumber = 1;
+var allValues = [];
+var numberOfPlayer = 2;
+var diceNumber = 2;
+var whoPlays = `player`;
+var winCondition = `highest value wins`;
+var diceNumber = 2;
+var allDiceRoll = [];
+var currentDice = 1;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (input == `instructions`) {
+    return instruction();
+  }
+  if (input == `computer` || input == `player`) {
+    whoPlays = input;
+    return `Your player has been set to ${input}.Press submit to start the game. For any queries, enter 'instructions' and press submit.`;
+  }
+
+  if (input == `highest value wins` || input == `lowest value wins`) {
+    winCondition = input;
+    return `Your winning criteria has been set to ${input}.Press submit to start the game. For any queries, enter 'instructions' and press submit.`;
+  }
+
+  if (mode == `instructions`) {
+    mode = `roll dice`;
+    return instruction();
+  }
+  if (input == `player number` || input == `dice number`) {
+    mode = input;
+    return `Your mode has been set to ${input}. Please input the ${input} you want`;
+  }
+  if (mode == `player number`) {
+    numberOfPlayer = input;
+    return `The number of players have been set to ${numberOfPlayer}. Press submit to start the game. For any queries, enter 'instructions' and press submit.`;
+  }
+  if (mode == `dice number`) {
+    diceNumber = input;
+    return `The number of dice have been set to ${diceNumber}. Press submit to start the game. For any queries, enter 'instructions' and press submit.`;
+  }
+  console.log(1);
+  while (playerNumber <= numberOfPlayer) {
+    console.log(2);
+    if (mode == `roll dice`) {
+      console.log(3);
+      while (currentDice <= diceNumber) {
+        indivisualDiceRoll = diceRoll();
+        allDiceRoll.push(indivisualDiceRoll);
+        console.log(4);
+        currentDice = currentDice + 1;
```

Comment:
sequence is clear, variables well named. its easy to follow the flow of the logic.
Nice use of console logs to track the sequence of the code